services/QuillLMS/app/lib/quill_authentication.rb

```diff
@@ -22,6 +22,37 @@ module QuillAuthentication
     )
   end
 
+  def admin!
```

Comment:
So this is kind of nit-picky, but technically, this is not `authentication` (even though we're putting it in a module called `QuillAuthentication`), it is `authorization`.  The fact that we have `current_user` means that the user has already authenticated, and now we're seeing if they're authorized for a resource.  This almost certainly doesn't matter, but it is sometimes useful to maintain a separation of the two concerns.

No suggested alternative, so this is fine really, I just wanted to flag it for possible future consideration.

Reply:
I was debating  a `QuillAuthorization` concern but opted not to.  I think your comment means I should.

Reply:
Although looking at it again I realized that the `auth_failed` method is kind of doing double duty.  I'll make a note to refactor this later.  It's definitely a good call out